cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -1790,8 +1816,10 @@ const char* lpad_utf8_int32_utf8(gdv_int64 context, const char* text, gdv_int32
     // case (return_length > text_char_count)
     // case where it needs to copy "fill_text" on the string left. The total number
     // of chars to copy is given by (return_length -  text_char_count)
-    char* ret =
-        reinterpret_cast<gdv_binary>(gdv_fn_context_arena_malloc(context, return_length));
+    gdv_int32 return_char_length = evaluate_return_char_length(
```

Comment:
return_char_length seems to be used only for allocation. Shouldn't the while loop in 1832 also use return_char_length?

Reply:
Changed to actual_text_len. @rkavanap if no more issues, could you help merge this pr since it's been quite a while since open. Thanks a lot.

Reply:
@ZMZ91 I am done with my review and this PR looks good. Please let me know what else I need to do to help merge the PR.

Reply:
What should I do next? I have no permission to merge this pr, right?

Reply:
Can you check with @projjal ?

Reply:
ALso you may have to rebase and make the failing checks pass